Automattic/jetpack
# General: update minimum required version to WordPress 5.5

#### Changes proposed in this Pull Request:

* Update minimum version required by the plugin.
* Update tested version.
* Remove fallbacks that were needed for WordPress 5.5-.

#### Jetpack product discussion

* Primary issue: #17064  

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

Start with a site running WP 5.5.

* Do the tests pass?
* Can you browse through Jetpack > Dashboard, Jetpack > Settings? Note the strings in use in the At A Glance cards in Jetpack > Dashboard, for example.
* Can you go to the block editor? Are the blocks accessible?
* Can you insert a Loom block? Do you see the variation available in the block editor when you search for it?
* Can you insert a form block? Check the strings in use in the block sidebar.

#### Proposed changelog entry for your changes:

* General: Jetpack now requires WordPress 5.5, and is fully compatible with WordPress 5.6.
